### PR TITLE
Fix DDL backout in the block processor

### PR DESCRIPTION
**Pull request by riverszhang89**
Ensure that when aborting a DDL transaction, its parent transaction is aborted too.

(DRQS 163967763)

### CODE REVIEW

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: Success &#10003;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/ec2d934/robomark.txt'>Error &#x26A0; </a>.<br>Regression testing: <b>1/372 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>sc_transactional_rowlocks_generated [core dumped]</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/ec2d934/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/ec2d934/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2510'>Resubmit</a></p>

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: Success &#10003;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/856df60/robomark.txt'>Error &#x26A0; </a>.<br>Regression testing: <b>2/372 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>sc_transactional_rowlocks_generated [core dumped]<br>socksql_master_swings</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/856df60/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/856df60/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2510'>Resubmit</a></p>

**Review by roborivers — COMMENTED** [review]
<p>Sysbench Transactions Per Second (commit: 856df60):</p>

<pre lang=diff>
@@ sysbench/oltp_read_only @@
+13.15%
</pre>

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: Success &#10003;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/725f685/robomark.txt'>Error &#x26A0; </a>.<br>Regression testing: <b>1/372 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>socksql_master_swings</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/725f685/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/725f685/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2510'>Resubmit</a></p>

File: db/toblock.c [hunk 1]
```diff
@@ -2400,7 +2400,9 @@ static void backout_and_abort_tranddl(struct ireq *iq, tran_type *parent,
     int rc = 0;
     if (rowlocks) {
         assert(parent);
-        rc = trans_abort(iq, bdb_get_physical_tran(parent));
+        /* Check if we started a physical transaction. */
+        if (bdb_get_physical_tran(parent) != NULL)
+            rc = trans_abort(iq, bdb_get_physical_tran(parent));
         bdb_reset_physical_tran(parent);
```
Comment (mistachkin) [inline db/toblock.c:2406]: The call to bdb_reset_physical_tran() should be inside the "if" block here.

Comment (riverszhang89) [reply db/toblock.c]: Thank you, Joe! I've moved the lines inside the if block.

**Review by nirbhayc — APPROVED** [review]
Looks good!

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: Success &#10003;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>1/372 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>sc_truncate_lockorder_generated [core dumped]</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/725f685/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/725f685/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2510'>Resubmit</a></p>

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: Success &#10003;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>3/372 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>socksql_master_swings<br>queuedb_rollover_noroll1_generated<br>logfill</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/2a27a73/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/2a27a73/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2510'>Resubmit</a></p>

**Comment by riverszhang89** [comment]
I've closely examined the test failures. None seems related to my changes.

**Comment by riverszhang89** [comment]
@mistachkin @nirbhayc Thank you for the review!

**Pull request merged**